lib/sensors/temperature.cpp

```diff
@@ -0,0 +1,53 @@
+#include "temperature.hpp"
+
+namespace hyped::sensors {
+
+Temperature::Temperature(hyped::io::I2c &i2c, hyped::core::ILogger &log) : log_(log), i2c_(i2c)
+{
+}
+
+std::optional<std::uint16_t> Temperature::read()
+{
+  const auto Status_check = i2c_.readByte(kMux, kStatus);
+
+  if (Status_check) {
+    if (Status_check.value() == 0) {
+      const std::optional<std::uint8_t> tempurature_high = i2c_.readByte(kMux, kTempuratureHigh);
+      const std::optional<std::uint8_t> tempurature_low  = i2c_.readByte(kMux, kTempuratureLow);
+      // getting the high and low values from the register
+
+      if (tempurature_high.has_value() && tempurature_low.has_value()) {
+        std::uint16_t tempurature = (tempurature_high.value() << 8) | tempurature_low.value();
+        tempurature               = tempurature * 0.01;
+
+        return tempurature;
+      } else {
+        log_.log(hyped::core::LogLevel::kFatal, "Temperature read not implemented");
+        return std::nullopt;
+      }
+      // Reading from the Status register to see if the tempurature is ready to be read
+      // Need to have it do the correct effect and output a log of what occurred
+
+    } else {
+      log_.log(hyped::core::LogLevel::kFatal, "Temperature read not implemented");
+      return std::nullopt;
+    }
+  } else {
+    log_.log(hyped::core::LogLevel::kFatal, "Temperature read not implemented");
+    return std::nullopt;
+  }
+}
```

Comment:
I dont think this is very readable. Start with checking individually if temp_high or temp_low has a value and log + return nullopt if not. Then you can straight away do the bit shifting part since you will have ensured both have a value.